collections/nemo_nlp/nemo_nlp/data/datasets/utils.py

```diff
@@ -312,6 +317,106 @@ def process_atis(infold, uncased, modes=['train', 'test'], dev_split=0):
     return outfold
 
 
+def generate_BIO(slot_tags_list, token_str, start_i, end_i):
```

Comment:
I don't think we need this function, but if we decide to keep this, there are a couple of things:

1. please add docstring to explain why we need this.
2. can end_i ever be greater than tag_end?

Reply:
Merged it into the caller function. end_i can be larger in cases that start_i is also larger than the tag_start.